test/Rentals.spec.ts

```diff
@@ -1732,6 +1801,73 @@ describe('Rentals', () => {
       expect(rental.endDate).to.equal(latestBlockTimestamp + daysToSeconds(offerParams.rentalDays))
     })
 
+    it('should increase the end date of a rental started by sending the asset to the contract, for an extension with accept offer', async () => {
+      const bytes = ethers.utils.defaultAbiCoder.encode([offerEncodeType], [offerEncodeValue])
+
+      await land.connect(lessor)['safeTransferFrom(address,address,uint256,bytes)'](lessor.address, rentals.address, tokenId, bytes)
+
+      const latestBlockTimestamp = await getLatestBlockTimestamp()
+
+      let endDate = latestBlockTimestamp + daysToSeconds(offerParams.rentalDays)
+
+      let rental = await rentals.rentals(offerParams.contractAddress, offerParams.tokenId)
+
+      expect(rental.endDate).to.equal(endDate)
+
+      const increaseTime = Math.trunc(daysToSeconds(offerParams.rentalDays) / 2)
+
+      await evmIncreaseTime(increaseTime)
+      await evmMine()
+
+      offerParams = { ...offerParams, nonces: [0, 0, 1], expiration: maxUint256 }
+
+      await rentals.connect(lessor).acceptOffer({ ...offerParams, signature: await getOfferSignature(tenant, rentals, offerParams) })
+
+      rental = await rentals.rentals(offerParams.contractAddress, offerParams.tokenId)
+
+      endDate += daysToSeconds(offerParams.rentalDays)
+
+      expect(rental.endDate).to.equal(endDate)
+    })
+
+    it('should increase the end date of a rental started by sending the asset to the contract, for an extension with accept listing', async () => {
+      const bytes = ethers.utils.defaultAbiCoder.encode([offerEncodeType], [offerEncodeValue])
+
+      await land.connect(lessor)['safeTransferFrom(address,address,uint256,bytes)'](lessor.address, rentals.address, tokenId, bytes)
+
+      const latestBlockTimestamp = await getLatestBlockTimestamp()
+
+      let endDate = latestBlockTimestamp + daysToSeconds(acceptListingParams.rentalDays)
+
+      let rental = await rentals.rentals(listingParams.contractAddress, listingParams.tokenId)
+
+      expect(rental.endDate).to.equal(endDate)
+
+      const increaseTime = Math.trunc(daysToSeconds(acceptListingParams.rentalDays) / 2)
+
+      await evmIncreaseTime(increaseTime)
+      await evmMine()
+
+      listingParams = { ...listingParams, nonces: [0, 0, 1], expiration: maxUint256 }
+      acceptListingParams = { ...acceptListingParams, operator: extra.address }
+
+      await rentals
+        .connect(tenant)
+        .acceptListing(
+          { ...listingParams, signature: await getListingSignature(lessor, rentals, listingParams) },
+          acceptListingParams.operator,
+          acceptListingParams.index,
+          acceptListingParams.rentalDays,
+          acceptListingParams.fingerprint
+        )
+
+      rental = await rentals.rentals(listingParams.contractAddress, listingParams.tokenId)
+
+      endDate += daysToSeconds(acceptListingParams.rentalDays)
+
+      expect(rental.endDate).to.equal(endDate)
+    })
```

Comment:
I would add more tests for revert cases like trying to extend the dates by not the lessor, etc

Reply:
Added various tests for extend revertions